charts/castai-spot-handler/templates/daemonset.yaml

```
@@ -26,6 +26,9 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
+      {{- if .Values.priorityClassName }}
```

```suggestion
     {{- if and (.Capabilities.APIVersions.Has "scheduling.k8s.io/v1") (semverCompare ">= 1.18-0" .Capabilities.KubeVersion.Version) .Values.priorityClassName }}
```
https://github.com/castai/helm-charts/blob/main/charts/castai-cluster-controller/templates/deployment.yaml#L54

I would copy-paste this condition to be safe